File: src/main/java/hu/noSignal/noSignal/Modell/Comment.java
package hu.noSignal.noSignal.Modell;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Table(name = "COMMENTS")
public class Comment extends BaseEntity {

    @Column(nullable = false)
    private long postid;

    @Column(nullable = false)
    private long userid;

    @Column(nullable = false)
    private Timestamp date;

    @Column(nullable = false)
    private String text;

    @Column
    private String username;

    @JsonIgnore
    @JoinColumn(name = "post", referencedColumnName = "id")
    @ManyToMany(targetEntity = Post.class)
    private List<Post> posts;
}

File: noSginal/src/app/model/Comment.ts
import {Timestamp} from "rxjs/Rx";

export class SignalComment {
  id: Number;
  userid: Number;
  postid: Number;
  date: Timestamp<Number>;
  text: string;
  username: String;

  constructor(userid?: Number, data?: string, postid?: Number, id?: Number,) {
    this.text = data;
    this.id = id || null;
    this.postid = postid;
    this.userid = userid || null;
  }
}
